src/components/svg/keys/svg-keyboard-key/svg-keyboard-key.component.ts

```diff
@@ -61,45 +71,122 @@ export class SvgKeyboardKeyComponent implements OnInit, OnChanges, OnDestroy {
     @Input() keyAction: KeyAction;
     @Input() keybindAnimationEnabled: boolean;
     @Output() keyClick = new EventEmitter();
+    @Output() captured = new EventEmitter();
 
     enumLabelTypes = LabelTypes;
 
+    public changeAnimation: string = 'inactive';
+    public recordAnimation: string;
+
     private labelSource: any;
     private labelType: LabelTypes;
     private macros: Macro[];
     private subscription: Subscription;
-    private animation: string = 'inactive';
+    private record: boolean;
 
-    @HostListener('click') onClick() {
+    @HostListener('click')
+    onClick() {
+        this.reset();
         this.keyClick.emit(this.element.nativeElement);
     }
 
-    constructor(private mapper: MapperService, private store: Store<AppState>, private element: ElementRef) {
+    @HostListener('mouseup', ['$event'])
+    onMouseUp(e: MouseEvent) {
+        if (e.which === 2 || e.button === 1) {
+            this.record = true;
+            this.recordAnimation = 'active';
+        }
+    }
+
+    @HostListener('keyup')
+    onKeyUp() {
+        if (this.record) {
+            this.saveScanCode();
+        }
+    }
+
+    @HostListener('keydown', ['$event'])
+    onKeyDown(e: KeyboardEvent) {
+        const code: number = e.keyCode;
+
+        if (this.record) {
+            e.preventDefault();
+
+            if (this.captureService.hasMap(code)) {
+                this.saveScanCode(this.captureService.getMap(code));
+            } else {
+                this.captureService.setModifier((e.location === 1), code);
+            }
+        }
+    }
+
+    @HostListener('focusout')
+    onFocusOut() {
+        this.reset();
+    }
+
+    constructor(
+        private mapper: MapperService,
+        private store: Store<AppState>,
+        private element: ElementRef,
+        private captureService: CaptureService
+    ) {
         this.subscription = store.let(getMacroEntities())
             .subscribe((macros: Macro[]) => this.macros = macros);
+
+        this.reset();
+        this.captureService.populateMapping();
     }
 
     ngOnInit() {
         this.setLabels();
     }
 
     ngOnChanges(changes: { [propertyName: string]: SimpleChange }) {
-        /* tslint:disable:no-string-literal */
         if (changes['keyAction']) {
             this.setLabels();
             if (this.keybindAnimationEnabled) {
-                this.animation = 'active';
+                this.changeAnimation = 'active';
             }
         }
-        /* tslint:enable:no-string-literal */
     }
 
     ngOnDestroy() {
         this.subscription.unsubscribe();
     }
 
-    animationDone() {
-        this.animation = 'inactive';
+    changeDone() {
+        this.changeAnimation = 'inactive';
+    }
+
+    recordingDone() {
```

Comment:
Rename to `onRecordingAnimationDone` or `onRecordingDone` depending which you prefer.